### PR TITLE
JAMES-2774 Avoid nested block in ReactorRabbitMQChannelPool

### PR DESCRIPTION
**Pull request by chibenwa**

### CODE REVIEW

**Comment by chibenwa** [comment]
Build output lost.

Testing :green_apple:

test this please

File: backends-common/rabbitmq/src/main/java/org/apache/james/backends/rabbitmq/ReactorRabbitMQChannelPool.java [hunk 1]
```diff
@@ -138,28 +138,22 @@ public Receiver createReceiver() {
...
+        return tryBorrowFromPool()
             .doOnError(throwable -> LOGGER.warn("Cannot borrow channel", throwable))
             .retryBackoff(MAX_BORROW_RETRIES, MIN_BORROW_DELAY, FOREVER, Schedulers.elastic())
             .onErrorMap(this::propagateException)
             .subscribeOn(Schedulers.elastic())
-            .block();
+            .doOnNext(borrowedChannels::add);
+    }
+
+    private Mono<Channel> tryBorrowFromPool() {
```
Comment (trantienduchn) [inline backends-common/rabbitmq/src/main/java/org/apache/james/backends/rabbitmq/ReactorRabbitMQChannelPool.java:155]: this method is used only one place. Can you remove it?

File: backends-common/rabbitmq/src/main/java/org/apache/james/backends/rabbitmq/ReactorRabbitMQChannelPool.java [hunk 2]
```diff
@@ -138,28 +138,22 @@ public Receiver createReceiver() {
...
 
-    private Channel borrow() {
-        Channel channel = tryBorrowFromPool();
-        borrowedChannels.add(channel);
-        return channel;
-    }
-
-    private Channel tryBorrowFromPool() {
-        return Mono.fromCallable(this::borrowFromPool)
+    private Mono<Channel> borrow() {
```
Comment (trantienduchn) [inline backends-common/rabbitmq/src/main/java/org/apache/james/backends/rabbitmq/ReactorRabbitMQChannelPool.java:146]: can you move it inside `getChannelMono()` ?

Comment (chibenwa) [reply backends-common/rabbitmq/src/main/java/org/apache/james/backends/rabbitmq/ReactorRabbitMQChannelPool.java:146]: I thought about it but refrained as the `borrow` naming is very helpful to actually explain what's happening.

File: backends-common/rabbitmq/src/main/java/org/apache/james/backends/rabbitmq/ReactorRabbitMQChannelPool.java (hunk 1 above)
Comment (chibenwa) [reply backends-common/rabbitmq/src/main/java/org/apache/james/backends/rabbitmq/ReactorRabbitMQChannelPool.java:155]: IMO no as it adds semantic.

**Comment by Arsnael** [comment]
CI restart

test this please

File: backends-common/rabbitmq/src/main/java/org/apache/james/backends/rabbitmq/ReactorRabbitMQChannelPool.java [hunk 3]
```diff
@@ -138,28 +138,22 @@ public Receiver createReceiver() {
...
+        return tryBorrowFromPool()
             .doOnError(throwable -> LOGGER.warn("Cannot borrow channel", throwable))
             .retryBackoff(MAX_BORROW_RETRIES, MIN_BORROW_DELAY, FOREVER, Schedulers.elastic())
```
Comment (mbaechler) [inline backends-common/rabbitmq/src/main/java/org/apache/james/backends/rabbitmq/ReactorRabbitMQChannelPool.java:149]: are we retrying whatever the exception?

Comment (chibenwa) [reply backends-common/rabbitmq/src/main/java/org/apache/james/backends/rabbitmq/ReactorRabbitMQChannelPool.java:149]: Agreed, but IMO it's an other concern. Can we keep it for another PR?

**Comment by chibenwa** [comment]
Merged

File: backends-common/rabbitmq/src/main/java/org/apache/james/backends/rabbitmq/ReactorRabbitMQChannelPool.java (hunk 3 above)
Comment (mbaechler) [reply backends-common/rabbitmq/src/main/java/org/apache/james/backends/rabbitmq/ReactorRabbitMQChannelPool.java:149]: Is it a question? I mean, you merged the PR before I can answer

Comment (chibenwa) [reply backends-common/rabbitmq/src/main/java/org/apache/james/backends/rabbitmq/ReactorRabbitMQChannelPool.java:149]: Sorry if I went a bit fast here.

The comment was unrelated to the change.

Also Duc is doing some testing related to this here: https://github.com/linagora/james-project/pull/3282?notification_referrer_id=MDE4Ok5vdGlmaWNhdGlvblRocmVhZDgxNjI0MDU4NDo2OTI4NzQw&notifications_query=is%3Adone

**Pull request closed without merging**